### DISCUSSION TITLE
Enforce single execution of scheduled job per environment?

### DISCUSSION THREAD

**Discussion by turacma** [medusajs/medusa#10750] [Q&A] [answered]
I've set up a scheduled job in my new medusa environment, but noticed that it runs on each node running the application. I thought maybe this was due to a misconfiguration in my event bus config where I was using in-memory, but even after correcting this and updating to redis, I'm still seeing the same behavior. Is this expected or did I misconfigure something?

**Comment by carlos-r-l-rodrigues** [comment, marked as answer]
Hi @turacma,
the in-memory will behave like this, but when using the Redis one, only one worker will execute it at time.

did you set up the workflow-engine-redis?

```js
modules: {
  [Modules.WORKFLOW_ENGINE]: {
    resolve: "@medusajs/workflow-engine-redis",
    options: {
      redis: {
        url: process.env.REDIS_URL,
      },
    },
  },
}
```

**Reply by turacma** [reply]
@carlos-r-l-rodrigues thanks, this is the configuration I was missing.
